pkg/smi/traffic_target.go

```diff
@@ -0,0 +1,168 @@
+	// Check if permissive mode is enabled, in which case every meshed pod is allowed to communicate with each other
+	if check.cfg.IsPermissiveTrafficPolicyMode() {
+		return outcomes.DiagnosticOutcome{LongDiagnostics: "OSM is in permissive traffic policy modes -- all meshed pods can communicate and SMI access policies are not applicable"}
+	}
+	trafficTargets, err := check.accessClient.AccessV1alpha3().TrafficTargets(check.dstPod.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return outcomes.FailedOutcome{Error: err}
+	}
+	var matchingTargetNames []string
+	for _, trafficTarget := range trafficTargets.Items {
+		if doesTargetMatchPods(trafficTarget.Spec, check.srcPod, check.dstPod) {
+			matchingTargetNames = append(matchingTargetNames, trafficTarget.Name)
+		}
+	}
+	if len(matchingTargetNames) > 0 {
+		return outcomes.DiagnosticOutcome{LongDiagnostics: fmt.Sprintf("Pod '%s/%s' is allowed to communicate to pod '%s/%s' via SMI TrafficTarget policy/policies %s\n",
+			check.srcPod.Namespace, check.srcPod.Name, check.dstPod.Namespace, check.dstPod.Name, strings.Join(matchingTargetNames, ", "))}
+	}
+	return outcomes.DiagnosticOutcome{LongDiagnostics: fmt.Sprintf("Pod '%s/%s' is not allowed to communicate to pod '%s/%s' via any SMI TrafficTarget policy\n",
+		check.srcPod.Namespace, check.srcPod.Name, check.dstPod.Namespace, check.dstPod.Name)}
+}
+
+// Suggestion implements common.Runnable
+func (check TrafficTargetCheck) Suggestion() string {
+	return fmt.Sprintf("Check that source and desintation pod are referred to in a TrafficTarget. To get relevant TrafficTargets, use: \"kubectl get traffictarget -n %s -o yaml\"", check.dstPod.Namespace)
+}
+
+// FixIt implements common.Runnable
+func (check TrafficTargetCheck) FixIt() error {
+	panic("implement me")
+}
+
+func doesTargetMatchPods(spec access.TrafficTargetSpec, srcPod *corev1.Pod, dstPod *corev1.Pod) bool {
+	// Map traffic targets to the given pods
+	if !cli.DoesTargetRefDstPod(spec, dstPod) {
+		return false
+	}
+	// The TrafficTarget destination is associated to 'dstPod', check if 'srcPod` is an allowed source to this destination
+	if cli.DoesTargetRefSrcPod(spec, srcPod) {
+		return true
+	}
+	return false
+}
+
+// Verify interface compliance
+var _ common.Runnable = (*RoutesValidityCheck)(nil)
+
+// RoutesValidityCheck implements common.Runnable
+type RoutesValidityCheck struct {
+	cfg          configurator.Configurator
+	srcPod       *corev1.Pod
+	dstPod       *corev1.Pod
+	accessClient smiAccessClient.Interface
+}
+
+// AreTrafficRoutesValid checks whether TrafficTargets matching the src and dest pods have supported routes
+func AreTrafficRoutesValid(osmConfigurator configurator.Configurator, srcPod *corev1.Pod, dstPod *corev1.Pod, smiAccessClient smiAccessClient.Interface) RoutesValidityCheck {
+	return RoutesValidityCheck{
+		cfg:          osmConfigurator,
+		srcPod:       srcPod,
+		dstPod:       dstPod,
+		accessClient: smiAccessClient,
+	}
+}
+
+const (
```

Comment:
@SanyaKochhar It would be great to carve this out into a separate file.

This is starting to form as a great pattern - each check (Runnable) could live in a file.
This file could be `httproutegroup_check.go` or something like that.